docs/proposals/scheduling/20240115-elastic-quota-support-job-level-oversold-management.md

```diff
@@ -0,0 +1,295 @@
+resources fairly to different quotas according to a certain proportion. However, in AI training scenario, due to job's
+"AllOrNothing" property, the runtime quota mechanism may lead to preemptible job can't be assigned successfully because
+each quota only obtains partial idle resources. So our strategy is that the idle resources will be assigned to each
+quota by a greedy way, which means if one quota's preemptible job comes first, it can use all idle resources as much as it reaches its max.
+
+#### Preemption
+
+##### Quota and Node Binding
+In AI training scenario, we usually bind quota and nodes together. some reasons are:
+1. AI training jobs have a large network communication and will be easily affected by network topology\quantity\speed etc.
+   In a shared cluster, due to resource competition reason, we can't assign the non-preemptible job with the best network 
+   topology for each assignment, which will lead to training speed unstable, and it is difficult for users to accept. Binding 
+   quota and nodes together will effectively alleviate this problem.
+2. In a shared cluster, we may often meet resource fragment problem. In AI training scenario, some job are large like need
+   128 worker and per worker need 8 gpu cards, meanwhile some job are small like need 1 worker and per worker need 1 gpu card.
+   The more small jobs, the more resource fragments to large jobs. Binding quota and nodes together will restrict the scope
+   of resource fragment within the quota, and users in one quota can communicate and coordinate to solve the fragment.
+
+Normally, cluster maintainer will select some nodes binding to quota, and non-preemptible pods can only run on these nodes.
+For preemptible pods, maintainer can choose some nodes belongs to self quota and other quotas, or even the whole cluster nodes
+to increase scheduling successful-probability. When non-preemptible pods need to schedule, it will preempt the preemptible
+pods those running on non-preemptible-quota-binding-nodes.
+
+We do not have specific requirements for the method of quota node binding. Cluster maintainer can use affinity\label to bind
+concrete nodes, or first patch related nodes with quota label, then fill pods with affinity, or use resourceFlavor to
+select some nodes for quota and fill pod needed inputs. For our practise, the cluster maintainer will use "patch related
+nodes with quota label, then fill pods with affinity" way, non-preemptible pods will only fill its own quota affinity,
+and preemptible pods will fill several quotas based on business workload needs.
+
+In the end, node and quota binding should not have affect to preemption, so the doc below only introduce the node and quota binding
+in user-view, the concrete preemption implementation has no connection with quota node binding concept.
+
+##### Job Level Preemption
+As mentioned above, in AI scenario, job has property of "AllOrNothing", which means the process of resource allocation
```

Comment:
There is no need to repeat concepts that have been explained clearly. If repetition occurs, it proves that the language or layout is not well organized. It is recommended to emphasize what Job-level Preemption is at the beginning of the Proposal, or to explain clearly in Motivation why there is a Job- level preemption.

Reply:
done